Fix test_linux_tester_container tests broken by --pull removal in PR #218

## Problem

Build 254 reef step `ci+release tooling tests [g3_s0]` **fails** because two unit tests in `ci/ray_ci/test_linux_tester_container.py` assert a docker build command structure that no longer matches the source code.

PR #218 (`d6d929a6c2`) removed `--pull` from the docker build command in `ci/ray_ci/linux_container.py` but did **not** update the test assertions.

### Failing Tests

1. `test_ray_installation` (line ~173)
2. `test_ray_installation_wheel` (line ~199)

Both fail with:
```
AssertionError: At index 2 diff: '--progress=plain' != '--pull'
```

### Root Cause

The source code in `ci/ray_ci/linux_container.py` now generates:
```python
["docker", "build", "--progress=plain", "-t", ...]
```

But the tests still expect the old command with `--pull`:
```python
["docker", "build", "--pull", "--progress=plain", "-t", ...]
```

### Fix

In `ci/ray_ci/test_linux_tester_container.py`:

1. **`test_ray_installation`** (~line 173): Remove `"--pull",` from the expected list. The assertion list changes from 15 to 14 elements.

2. **`test_ray_installation_wheel`** (~line 199): Change `cmd[0:6]` to `cmd[0:5]` and remove `"--pull",` from the expected list:
   ```python
   assert cmd[0:5] == [
       "docker",
       "build",
       "--progress=plain",
       "-t",
       docker_image,
   ]
   ```

### Files to Modify

- `ci/ray_ci/test_linux_tester_container.py`

### Verification

Run the test locally inside the forge container:
```bash
bazel test //ci/ray_ci:test_linux_tester_container --config=ci
```

All 9 tests should pass (7 already pass, the 2 fixed tests should now also pass).

### Impact

This is the **only** failure in the reef test step — 24 of 25 Bazel test targets in that step passed. Fixing this will make g3_s0 pass, which is a Tier 3 milestone.

### Evidence

Build 254 logs: `019ce610-0db8-4083-99e4-07cb3455021b`
- Line 858: `AssertionError: At index 2 diff: '--progress=plain' != '--pull'`
- Line 1127: `//ci/ray_ci:test_linux_tester_container FAILED in 3 out of 3 in 1.2s`

--- a/ci/ray_ci/test_linux_tester_container.py
+++ b/ci/ray_ci/test_linux_tester_container.py
@@ -170,7 +170,6 @@ def test_ray_installation() -> None:
         assert install_ray_cmds[-1] == [
             "docker",
             "build",
-            "--pull",
             "--progress=plain",
             "-t",
             docker_image,
@@ -196,10 +195,9 @@ def test_ray_installation_wheel() -> None:
         LinuxTesterContainer("team", build_type="wheel", python_version="3.10")
         docker_image = f"{_DOCKER_WORK_REPO}:team"
         cmd = install_ray_cmds[-1]
-        assert cmd[0:6] == [
+        assert cmd[0:5] == [
             "docker",
             "build",
-            "--pull",
             "--progress=plain",
             "-t",
             docker_image,

